File: com/RaghavStaadExtractor/RaghavDatabase.vb
Imports System.Runtime.InteropServices
Imports Microsoft.Office.Interop.Excel

Namespace RaghavStaadExtractor

    <ComVisible(True)>
    <Guid("D2A2B1E4-44C2-41C3-9A54-9A48F77D1234")> ' Replace with your own GUID
    <ProgId("RaghavStaadExtractor.RaghavDatabase")>
    Public Class RaghavDatabase

        Private Const pwd As String = "2022"

        ' === Unlock sheets for data review ===
        <ComVisible(True)>
        Public Sub DataRewiewUnlocked()
            Dim app As Application = Nothing
            Try
                app = CType(Marshal.GetActiveObject("Excel.Application"), Application)
                Dim currentSheet As Worksheet = app.ActiveSheet
                Dim wb As Workbook = app.ActiveWorkbook

                app.ScreenUpdating = False
                app.EnableEvents = False
                app.Calculation = XlCalculation.xlCalculationManual

                wb.Unprotect(pwd)

                For i As Integer = 2 To wb.Sheets.Count
                    Dim ws As Worksheet = CType(wb.Sheets(i), Worksheet)
                    ws.Unprotect(pwd)
                    ws.Visible = XlSheetVisibility.xlSheetVisible
                    ws.Cells.Locked = True
                    ws.Protect(Password:=pwd,
                               AllowFormattingCells:=False,
                               AllowDeletingRows:=False,
                               AllowInsertingRows:=False,
                               AllowSorting:=False,
                               AllowFiltering:=False)
                Next

                wb.Protect(Password:=pwd, Structure:=True)
                currentSheet.Activate()

                app.ScreenUpdating = True
                MsgBox("Step Inside: The Data Room is Now Open.", vbInformation)
            Catch ex As Exception
                MsgBox("Error in DataRewiewUnlocked: " & ex.Message, vbCritical)
            Finally
                If app IsNot Nothing Then
                    app.ScreenUpdating = True
                    app.EnableEvents = True
                    app.Calculation = XlCalculation.xlCalculationAutomatic
                End If
            End Try
        End Sub

        ' === Generate Unit Weight Database ===
        <ComVisible(True)>
        Public Sub LetsGenrateUnitWt()
            Dim app As Application = Nothing
            Try
                app = CType(Marshal.GetActiveObject("Excel.Application"), Application)
                Dim wb As Workbook = app.ActiveWorkbook
                Dim wsDest As Worksheet = CType(wb.Sheets("Sheet1"), Worksheet)
                Dim wsSrc As Worksheet
                Dim pasteRow As Long = 6
                Dim serialNum As Long = 1

                app.ScreenUpdating = False
                app.EnableEvents = False
                app.Calculation = XlCalculation.xlCalculationManual

                wb.Unprotect(pwd)
                For i As Integer = 1 To wb.Sheets.Count
                    Dim ws As Worksheet = CType(wb.Sheets(i), Worksheet)
                    ws.Visible = XlSheetVisibility.xlSheetVisible
                    ws.Unprotect(pwd)
                Next

                wsDest.Cells.Clear()

                With wsDest.Range("A1:E1")
                    .Merge()
                    .Value = "RAGHAV DATABASE"
                    .HorizontalAlignment = XlHAlign.xlHAlignCenter
                    .VerticalAlignment = XlVAlign.xlVAlignCenter
                    .Font.Bold = True
                    .Font.Size = 14
                    .Interior.Color = RGB(247, 199, 172)
                End With

                wsDest.Range("A2:E3").Merge()
                wsDest.Range("A2:E3").Value = ""
                wsDest.Range("A4:E4").Merge()
                wsDest.Range("A4:E4").Interior.Color = RGB(247, 199, 172)

                With wsDest.Range("A5:E5")
                    .Value = {"S.No", "Name", "Staad Name", "AREA Ax(cm2)", "Kg/M"}
                    .Font.Bold = True
                    .Font.Color = RGB(0, 0, 0)
                    .Interior.Color = RGB(173, 216, 230)
                End With

                wsDest.Range("A1:E4").Borders.LineStyle = XlLineStyle.xlContinuous

                For i As Integer = 2 To wb.Sheets.Count
                    wsSrc = CType(wb.Sheets(i), Worksheet)
                    Dim lastRowSrc As Long = wsSrc.Cells(wsSrc.Rows.Count, "B").End(XlDirection.xlUp).Row

                    If lastRowSrc >= 3 Then
                        Dim rngToCopy As Range = wsSrc.Range("B3:D" & lastRowSrc)
                        wsDest.Cells(pasteRow, "B").Resize(rngToCopy.Rows.Count, 3).Value = rngToCopy.Value

                        For r As Long = pasteRow To pasteRow + rngToCopy.Rows.Count - 1
                            wsDest.Cells(r, "A").Value = serialNum
                            If IsNumeric(wsDest.Cells(r, "D").Value) Then
                                wsDest.Cells(r, "E").Value = Math.Round(CDbl(wsDest.Cells(r, "D").Value) * 0.785, 3)
                            Else
                                wsDest.Cells(r, "E").Value = "N/A"
                            End If
                            serialNum += 1
                        Next

                        pasteRow = wsDest.Cells(wsDest.Rows.Count, "B").End(XlDirection.xlUp).Row + 1
                    End If
                Next

                Dim finalRow As Long = wsDest.Cells(wsDest.Rows.Count, "A").End(XlDirection.xlUp).Row
                wsDest.Range("A6:E" & finalRow).Borders.LineStyle = XlLineStyle.xlContinuous
                wsDest.Columns("A:E").AutoFit()

                For i As Integer = 2 To wb.Sheets.Count
                    Dim ws As Worksheet = CType(wb.Sheets(i), Worksheet)
                    ws.Visible = XlSheetVisibility.xlSheetVeryHidden
                    ws.Protect(Password:=pwd,
                               AllowFormattingCells:=False,
                               AllowDeletingRows:=False,
                               AllowInsertingRows:=False,
                               AllowSorting:=False,
                               AllowFiltering:=False)
                Next

                wsDest.Unprotect(pwd)
                wsDest.Cells.Locked = False
                wb.Protect(pwd, Structure:=True)

                app.ScreenUpdating = True
                MsgBox("All set! PlateNova can now use the Raghav Database.", vbInformation)

            Catch ex As Exception
                MsgBox("Error in LetsGenrateUnitWt: " & ex.Message, vbCritical)
            Finally
                If app IsNot Nothing Then
                    app.ScreenUpdating = True
                    app.EnableEvents = True
                    app.Calculation = XlCalculation.xlCalculationAutomatic
                End If
            End Try
        End Sub

        ' === Unprotect sheets but allow adding rows ===
        <ComVisible(True)>
        Public Sub UnprotectSheetsAllowAdd()
            Dim app As Application = Nothing
            Try
                app = CType(Marshal.GetActiveObject("Excel.Application"), Application)
                Dim wb As Workbook = app.ActiveWorkbook
                Dim currentSheet As Worksheet = app.ActiveSheet

                app.ScreenUpdating = False
                app.EnableEvents = False
                app.Calculation = XlCalculation.xlCalculationManual

                wb.Unprotect(pwd)

                For Each ws As Worksheet In wb.Worksheets
                    ws.Visible = XlSheetVisibility.xlSheetVisible
                    ws.Unprotect(pwd)
                    ws.Protect(Password:=pwd,
                               AllowFormattingCells:=True,
                               AllowInsertingRows:=True,
                               AllowDeletingRows:=True,
                               AllowSorting:=True,
                               AllowFiltering:=True)
                Next

                currentSheet.Activate()
                app.ScreenUpdating = True
                MsgBox("Blank pages await your brilliance — go ahead, add those sheets and let ideas flow!", vbInformation)
            Catch ex As Exception
                MsgBox("Error in UnprotectSheetsAllowAdd: " & ex.Message, vbCritical)
            Finally
                If app IsNot Nothing Then
                    app.ScreenUpdating = True
                    app.EnableEvents = True
                    app.Calculation = XlCalculation.xlCalculationAutomatic
                End If
            End Try
        End Sub

    End Class
End Namespace

File: com/RaghavStaadExtractor_decompiled_fresh/RaghavDatabase.cs
// ===== RaghavStaadExtractor.RaghavStaadExtractor.RaghavDatabase =====

// Decompiled member-by-member from RaghavStaadExtractor.dll



private const string pwd = "2022";


public RaghavDatabase()
{
}


using System;
using System.Reflection;
using System.Runtime.CompilerServices;
using System.Runtime.InteropServices;
using Microsoft.Office.Interop.Excel;
using Microsoft.VisualBasic;
using Microsoft.VisualBasic.CompilerServices;

[ComVisible(true)]
public void DataRewiewUnlocked()
{
	Application application = null;
	try
	{
		application = (Application)Marshal.GetActiveObject("Excel.Application");
		Worksheet worksheet = (Worksheet)application.ActiveSheet;
		Workbook activeWorkbook = application.ActiveWorkbook;
		application.ScreenUpdating = false;
		application.EnableEvents = false;
		application.Calculation = XlCalculation.xlCalculationManual;
		activeWorkbook.Unprotect("2022");
		int count = activeWorkbook.Sheets.Count;
		for (int i = 2; i <= count; i = checked(i + 1))
		{
			Worksheet worksheet2 = (Worksheet)activeWorkbook.Sheets[i];
			worksheet2.Unprotect("2022");
			worksheet2.Visible = XlSheetVisibility.xlSheetVisible;
			worksheet2.Cells.Locked = true;
			worksheet2.Protect("2022", RuntimeHelpers.GetObjectValue(Missing.Value), RuntimeHelpers.GetObjectValue(Missing.Value), RuntimeHelpers.GetObjectValue(Missing.Value), RuntimeHelpers.GetObjectValue(Missing.Value), false, RuntimeHelpers.GetObjectValue(Missing.Value), RuntimeHelpers.GetObjectValue(Missing.Value), RuntimeHelpers.GetObjectValue(Missing.Value), false, RuntimeHelpers.GetObjectValue(Missing.Value), RuntimeHelpers.GetObjectValue(Missing.Value), false, false, false, RuntimeHelpers.GetObjectValue(Missing.Value));
		}
		activeWorkbook.Protect("2022", true, RuntimeHelpers.GetObjectValue(Missing.Value));
		worksheet.Activate();
		application.ScreenUpdating = true;
		Interaction.MsgBox("Step Inside: The Data Room is Now Open.", MsgBoxStyle.Information);
	}
	catch (Exception ex)
	{
		ProjectData.SetProjectError(ex);
		Exception ex2 = ex;
		Interaction.MsgBox("Error in DataRewiewUnlocked: " + ex2.Message, MsgBoxStyle.Critical);
		ProjectData.ClearProjectError();
	}
	finally
	{
		if (application != null)
		{
			application.ScreenUpdating = true;
			application.EnableEvents = true;
			application.Calculation = XlCalculation.xlCalculationAutomatic;
		}
	}
}


using System;
using System.Reflection;
using System.Runtime.CompilerServices;
using System.Runtime.InteropServices;
using Microsoft.Office.Interop.Excel;
using Microsoft.VisualBasic;
using Microsoft.VisualBasic.CompilerServices;

[ComVisible(true)]
public void LetsGenrateUnitWt()
{
	Application application = null;
	checked
	{
		try
		{
			application = (Application)Marshal.GetActiveObject("Excel.Application");
			Workbook activeWorkbook = application.ActiveWorkbook;
			Worksheet worksheet = (Worksheet)activeWorkbook.Sheets["Sheet1"];
			long num = 6L;
			long num2 = 1L;
			application.ScreenUpdating = false;
			application.EnableEvents = false;
			application.Calculation = XlCalculation.xlCalculationManual;
			activeWorkbook.Unprotect("2022");
			int count = activeWorkbook.Sheets.Count;
			for (int i = 1; i <= count; i++)
			{
				Worksheet worksheet2 = (Worksheet)activeWorkbook.Sheets[i];
				worksheet2.Visible = XlSheetVisibility.xlSheetVisible;
				worksheet2.Unprotect("2022");
			}
			worksheet.Cells.Clear();
			Range range = ((_Worksheet)worksheet).get_Range((object)"A1:E1", RuntimeHelpers.GetObjectValue(Missing.Value));
			range.Merge(RuntimeHelpers.GetObjectValue(Missing.Value));
			range.set_Value(RuntimeHelpers.GetObjectValue(Missing.Value), (object)"RAGHAV DATABASE");
			range.HorizontalAlignment = XlHAlign.xlHAlignCenter;
			range.VerticalAlignment = XlVAlign.xlVAlignCenter;
			range.Font.Bold = true;
			range.Font.Size = 14;
			range.Interior.Color = Information.RGB(247, 199, 172);
			range = null;
			((_Worksheet)worksheet).get_Range((object)"A2:E3", RuntimeHelpers.GetObjectValue(Missing.Value)).Merge(RuntimeHelpers.GetObjectValue(Missing.Value));
			((_Worksheet)worksheet).get_Range((object)"A2:E3", RuntimeHelpers.GetObjectValue(Missing.Value)).set_Value(RuntimeHelpers.GetObjectValue(Missing.Value), (object)"");
			((_Worksheet)worksheet).get_Range((object)"A4:E4", RuntimeHelpers.GetObjectValue(Missing.Value)).Merge(RuntimeHelpers.GetObjectValue(Missing.Value));
			((_Worksheet)worksheet).get_Range((object)"A4:E4", RuntimeHelpers.GetObjectValue(Missing.Value)).Interior.Color = Information.RGB(247, 199, 172);
			Range range2 = ((_Worksheet)worksheet).get_Range((object)"A5:E5", RuntimeHelpers.GetObjectValue(Missing.Value));
			range2.set_Value(RuntimeHelpers.GetObjectValue(Missing.Value), (object)new string[5] { "S.No", "Name", "Staad Name", "AREA Ax(cm2)", "Kg/M" });
			range2.Font.Bold = true;
			range2.Font.Color = Information.RGB(0, 0, 0);
			range2.Interior.Color = Information.RGB(173, 216, 230);
			range2 = null;
			((_Worksheet)worksheet).get_Range((object)"A1:E4", RuntimeHelpers.GetObjectValue(Missing.Value)).Borders.LineStyle = XlLineStyle.xlContinuous;
			int count2 = activeWorkbook.Sheets.Count;
			for (int j = 2; j <= count2; j++)
			{
				Worksheet worksheet3 = (Worksheet)activeWorkbook.Sheets[j];
				long num3 = Conversions.ToLong(NewLateBinding.LateGet(NewLateBinding.LateGet(worksheet3.Cells[worksheet3.Rows.Count, "B"], null, "End", new object[1] { XlDirection.xlUp }, null, null, null), null, "Row", new object[0], null, null, null));
				if (num3 < 3)
				{
					continue;
				}
				Range range3 = ((_Worksheet)worksheet3).get_Range((object)("B3:D" + Conversions.ToString(num3)), RuntimeHelpers.GetObjectValue(Missing.Value));
				NewLateBinding.LateSetComplex(NewLateBinding.LateGet(worksheet.Cells[num, "B"], null, "Resize", new object[2]
				{
					range3.Rows.Count,
					3
				}, null, null, null), null, "Value", new object[1] { range3.get_Value(RuntimeHelpers.GetObjectValue(Missing.Value)) }, null, null, OptimisticSet: false, RValueBase: true);
				long num4 = num;
				long num5 = num + range3.Rows.Count - 1;
				for (long num6 = num4; num6 <= num5; num6++)
				{
					NewLateBinding.LateSetComplex(worksheet.Cells[num6, "A"], null, "Value", new object[1] { num2 }, null, null, OptimisticSet: false, RValueBase: true);
					if (Versioned.IsNumeric(RuntimeHelpers.GetObjectValue(NewLateBinding.LateGet(worksheet.Cells[num6, "D"], null, "Value", new object[0], null, null, null))))
					{
						NewLateBinding.LateSetComplex(worksheet.Cells[num6, "E"], null, "Value", new object[1] { Math.Round(Conversions.ToDouble(NewLateBinding.LateGet(worksheet.Cells[num6, "D"], null, "Value", new object[0], null, null, null)) * 0.785, 3) }, null, null, OptimisticSet: false, RValueBase: true);
					}
					else
					{
						NewLateBinding.LateSetComplex(worksheet.Cells[num6, "E"], null, "Value", new object[1] { "N/A" }, null, null, OptimisticSet: false, RValueBase: true);
					}
					num2++;
				}
				num = Conversions.ToLong(Operators.AddObject(NewLateBinding.LateGet(NewLateBinding.LateGet(worksheet.Cells[worksheet.Rows.Count, "B"], null, "End", new object[1] { XlDirection.xlUp }, null, null, null), null, "Row", new object[0], null, null, null), 1));
			}
			long num7 = Conversions.ToLong(NewLateBinding.LateGet(NewLateBinding.LateGet(worksheet.Cells[worksheet.Rows.Count, "A"], null, "End", new object[1] { XlDirection.xlUp }, null, null, null), null, "Row", new object[0], null, null, null));
			((_Worksheet)worksheet).get_Range((object)("A6:E" + Conversions.ToString(num7)), RuntimeHelpers.GetObjectValue(Missing.Value)).Borders.LineStyle = XlLineStyle.xlContinuous;
			NewLateBinding.LateCall(worksheet.Columns["A:E", RuntimeHelpers.GetObjectValue(Missing.Value)], null, "AutoFit", new object[0], null, null, null, IgnoreReturn: true);
			int count3 = activeWorkbook.Sheets.Count;
			for (int k = 2; k <= count3; k++)
			{
				Worksheet worksheet4 = (Worksheet)activeWorkbook.Sheets[k];
				worksheet4.Visible = XlSheetVisibility.xlSheetVeryHidden;
				worksheet4.Protect("2022", RuntimeHelpers.GetObjectValue(Missing.Value), RuntimeHelpers.GetObjectValue(Missing.Value), RuntimeHelpers.GetObjectValue(Missing.Value), RuntimeHelpers.GetObjectValue(Missing.Value), false, RuntimeHelpers.GetObjectValue(Missing.Value), RuntimeHelpers.GetObjectValue(Missing.Value), RuntimeHelpers.GetObjectValue(Missing.Value), false, RuntimeHelpers.GetObjectValue(Missing.Value), RuntimeHelpers.GetObjectValue(Missing.Value), false, false, false, RuntimeHelpers.GetObjectValue(Missing.Value));
			}
			worksheet.Unprotect("2022");
			worksheet.Cells.Locked = false;
			activeWorkbook.Protect("2022", true, RuntimeHelpers.GetObjectValue(Missing.Value));
			application.ScreenUpdating = true;
			Interaction.MsgBox("All set! PlateNova can now use the Raghav Database.", MsgBoxStyle.Information);
		}
		catch (Exception ex)
		{
			ProjectData.SetProjectError(ex);
			Exception ex2 = ex;
			Interaction.MsgBox("Error in LetsGenrateUnitWt: " + ex2.Message, MsgBoxStyle.Critical);
			ProjectData.ClearProjectError();
		}
		finally
		{
			if (application != null)
			{
				application.ScreenUpdating = true;
				application.EnableEvents = true;
				application.Calculation = XlCalculation.xlCalculationAutomatic;
			}
		}
	}
}


using System;
using System.Collections;
using System.Reflection;
using System.Runtime.CompilerServices;
using System.Runtime.InteropServices;
using Microsoft.Office.Interop.Excel;
using Microsoft.VisualBasic;
using Microsoft.VisualBasic.CompilerServices;

[ComVisible(true)]
public void UnprotectSheetsAllowAdd()
{
	Application application = null;
	try
	{
		application = (Application)Marshal.GetActiveObject("Excel.Application");
		Workbook activeWorkbook = application.ActiveWorkbook;
		Worksheet worksheet = (Worksheet)application.ActiveSheet;
		application.ScreenUpdating = false;
		application.EnableEvents = false;
		application.Calculation = XlCalculation.xlCalculationManual;
		activeWorkbook.Unprotect("2022");
		IEnumerator enumerator = default(IEnumerator);
		try
		{
			enumerator = activeWorkbook.Worksheets.GetEnumerator();
			while (enumerator.MoveNext())
			{
				Worksheet worksheet2 = (Worksheet)enumerator.Current;
				worksheet2.Visible = XlSheetVisibility.xlSheetVisible;
				worksheet2.Unprotect("2022");
				worksheet2.Protect("2022", RuntimeHelpers.GetObjectValue(Missing.Value), RuntimeHelpers.GetObjectValue(Missing.Value), RuntimeHelpers.GetObjectValue(Missing.Value), RuntimeHelpers.GetObjectValue(Missing.Value), true, RuntimeHelpers.GetObjectValue(Missing.Value), RuntimeHelpers.GetObjectValue(Missing.Value), RuntimeHelpers.GetObjectValue(Missing.Value), true, RuntimeHelpers.GetObjectValue(Missing.Value), RuntimeHelpers.GetObjectValue(Missing.Value), true, true, true, RuntimeHelpers.GetObjectValue(Missing.Value));
			}
		}
		finally
		{
			if (enumerator is IDisposable)
			{
				(enumerator as IDisposable).Dispose();
			}
		}
		worksheet.Activate();
		application.ScreenUpdating = true;
		Interaction.MsgBox("Blank pages await your brilliance — go ahead, add those sheets and let ideas flow!", MsgBoxStyle.Information);
	}
	catch (Exception ex)
	{
		ProjectData.SetProjectError(ex);
		Exception ex2 = ex;
		Interaction.MsgBox("Error in UnprotectSheetsAllowAdd: " + ex2.Message, MsgBoxStyle.Critical);
		ProjectData.ClearProjectError();
	}
	finally
	{
		if (application != null)
		{
			application.ScreenUpdating = true;
			application.EnableEvents = true;
			application.Calculation = XlCalculation.xlCalculationAutomatic;
		}
	}
}
